tests/routes/socialAuth.test.js

```diff
@@ -0,0 +1,270 @@
+/* eslint-disable no-unused-expressions */
+import chai, { should } from 'chai';
+import chaiHttp from 'chai-http';
+import { config } from 'dotenv';
+import axios from 'axios';
+import sinon from 'sinon';
+import sinonChai from 'sinon-chai';
+import { OAuth2Client } from 'google-auth-library';
+
+import db from '../../src/models';
+import app from '../../src';
+import { performCallback, createOrFindUser } from '../../src/services/auth';
+import authController from '../../src/controllers/auth';
+
+chai.use(chaiHttp);
+chai.use(sinonChai);
+should();
+config();
+
+const { twitterLogin } = authController;
+
+before(async () => {
+  try {
+    await db.sync({ force: true });
+  } catch (error) {
+    return error;
+  }
+});
+
+describe('Social Login Route', () => {
+  it('should throw error social provider is invalid', async () => {
+    const response = await chai.request(app).post('/api/v1/auth/something');
+
+    response.should.have.status(400);
+    response.body.status.should.eql('error');
+  });
+
+  it('should throw error if access token is invalid', async () => {
+    const invalidToken = { accessToken: 'invalid access token' };
+    const response = await chai
+      .request(app)
+      .post('/api/v1/auth/google')
+      .send(invalidToken);
+
+    response.should.have.status(500);
+    response.body.status.should.eql('error');
+  });
+
+  describe('facebook Authentication method', () => {
+    let facebook;
+    let sampleToken;
+
+    beforeEach(() => {
+      facebook = sinon.stub(axios, 'get');
+      facebook.onFirstCall().returns({
+        data: {
+          access_token: '472743220180658|Ybiad9C30YZmY42mAWU0zFzgZkM',
+          token_type: 'bearer'
+        }
+      });
+      sampleToken = { accessToken: 'invalid access token' };
+    });
+
+    afterEach(() => {
+      axios.get.restore();
+    });
+
+    it('should throw error if access token is invalid', async () => {
+      facebook.onSecondCall().returns({
+        data: {
+          data: {
+            error: { code: 190, message: 'Invalid OAuth access token.' },
+            is_valid: false,
+            scopes: []
+          }
+        }
+      });
+
+      const response = await chai
+        .request(app)
+        .post('/api/v1/auth/facebook')
+        .send(sampleToken);
+
+      facebook.should.calledTwice;
+      response.should.have.status(500);
+      response.body.status.should.eql('error');
+    });
+
+    it('should throw error if facebook access token is invalid', async () => {
+      facebook.onSecondCall().returns({
+        data: {
+          data: {
+            app_id: '472743220180658',
```

Comment:
Please put all these data (throughout the file) into a test fixture and import it here (as done in other tests). It makes the code look cleaner e.g.